### PR TITLE
OCPBUGS-23544: Improving performance of creating VMs in Azure

### PR DESCRIPTION
**Pull request by JoelSpeed**
This will enable us to change the number of concurrent reconciles so that we can test the behaviour of the controller when we increase the number of reconciles from the default value of 1.

It also ensures that we set the `networkAddresses` and `providerID` fields in the status when the VM is created, rather than waiting for the next reconcile to update these. This is important because the network addresses not being set causes the CSR approver not to work, which, when scaling up in large quantities, can cause a backlog which eventually prevents any CSRs being approved. This change should mitigate those slow approvals.

### CODE REVIEW

**Comment by JoelSpeed** [comment]
Some hack in vendor for now for testing

File: pkg/cloud/azure/actuators/machine/actuator_test.go [hunk 1]
```diff
@@ -343,7 +343,7 @@ func TestReconcileVMSuceededState(t *testing.T) {
 		t.Errorf("failed to create machine: %+v", err)
 	}
 
-	if fakeVMService.GetCallCount != 1 {
+	if fakeVMService.GetCallCount != 2 {
 		t.Errorf("expected get to be called just once")
```
Comment (theobarberbany) [inline pkg/cloud/azure/actuators/machine/actuator_test.go:347]: called twice? 

To check I'm following: this is to reflect the create + update call, right?

**Comment by JoelSpeed** [comment]
/retest

Issues do not appear to be related

**Pull request merged**